packages/core/src/utils/viaProxy.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * Utility to redirect remote sources via Proxy
+ *
+ * This utility is used to rewrite a request to be routed through
+ * the Proxy instead.
+ */
+
+/**
+ * Virtual Module provided by the Motion Canvas
+ * Plugin. Returns `false` if proxy is set to `false`, else `true`
+ */
+//@ts-ignore
+import {enabled} from 'virtual:proxysettings';
+
+/**
+ * Helper-Function to route remote requests through a local proxy.
+ *
+ * @example This rewrites a remote url like `https://via.placeholder.com/300.png/09f/fff` into a URI-Component-Encoded string like `/cors-proxy/https%3A%2F%2Fvia.placeholder.com%2F300.png%2F09f%2Ffff`
+ */
+export function viaProxy(url: string, overrideProxySetting?: boolean) {
```

Comment:
I had to add `overrideProxySetting` for Unit Tests here. I could not manage to mock the virtual module properly.

It would only ever consider the first mocked value and ignore subsequent mocks. If you have an idea on how to get the mocks to update, please do let me know.

Reply:
no longer relevant